src/brpc/rdma/rdma_endpoint.cpp

```diff
@@ -1032,11 +1070,20 @@ int RdmaEndpoint::PostRecv(uint32_t num, bool zerocopy) {
             } else {
                 CHECK(static_cast<uint32_t>(size) == g_rdma_recv_block_size) << size;
             }
+#endif  // if BRPC_WITH_GDR
         }
+#if BRPC_WITH_GDR
+        if (DoPostRecvGDR(_rbuf_data[_rq_received], g_rdma_recv_block_size, lkey) < 0) {
+            _rbuf[_rq_received].clear();
+            return -1;
+        }
+#else
         if (DoPostRecv(_rbuf_data[_rq_received], g_rdma_recv_block_size) < 0) {
             _rbuf[_rq_received].clear();
             return -1;
         }
```

Comment:
There are two posts here, which should be the reason why #3145 with GDR has too many IMMs.